.github/workflows/scan.yaml

```diff
@@ -0,0 +1,140 @@
+          done <<< "${supported_releases}"
+
+          echo "DEBUG: ${tags}"
+
+          # build a JSON formatted list of all the supported releases for crossplane/crossplane
+          supported_releases=$(echo $supported_releases | jq -R .| jq -s -c '.[] | split(" ")')
+          ## build a map of all the supported releases and their latest tags for later usage
+          versions=$(echo $tags | jq -R .| jq -s -c '.[] | split(" ") | [.[] | select(length > 0) | [split("=")] | map({key: .[0], value: .[1]}) | .[] ] | from_entries' )
+
+          # store everything as outputs
+          echo "versions=${versions}" >> $GITHUB_OUTPUT
+          echo "supported_releases=${supported_releases}" >> $GITHUB_OUTPUT
+
+          echo "DEBUG: GITHUB_OUTPUT:"
+          cat $GITHUB_OUTPUT
+
+  check-matrix:
+    # this job is just to check the matrix definition is valid and helps debugging it if not valid
+    runs-on: ubuntu-latest
+    needs:
+      - generate-matrix
+    steps:
+      - name: Check Matrix Definition
+        shell: bash
+        run: |
+          supported_releases='${{ needs.generate-matrix.outputs.supported_releases }}'
+          echo $supported_releases
+          echo $supported_releases | jq .
+
+  scan:
+    needs:
+      - check-matrix
+      - generate-matrix
+    strategy:
+      fail-fast: false
+      matrix:
+        release: ${{ fromJSON(needs.generate-matrix.outputs.supported_releases) }}
+        image:
+          - crossplane/crossplane
+          - crossplane/xfn
+        exclude:
+          # excluded because xfn was introduced only in v1.11
+          - image: crossplane/xfn
+            release: v1.9
+          - image: crossplane/xfn
+            release: v1.10
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get Release Tag
+        run: |
+          echo "${{ matrix.release }}"
+          tag="$(echo '${{ needs.generate-matrix.outputs.versions }}' | jq --raw-output ".[\"${{ matrix.release }}\"]")"
+          echo "tag=${tag}" >> $GITHUB_ENV
+          echo "escaped_filename=$(echo ${{ matrix.image }}/$tag | sed 's/[\/.:]/_/g')" >> $GITHUB_ENV
+
+      # we log to DockerHub to avoid rate limiting
+      - name: Login To DockerHub
+        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a # v2
+        if: env.DOCKER_USR != ''
+        with:
+          username: ${{ secrets.DOCKER_USR }}
+          password: ${{ secrets.DOCKER_PSW }}
+
+      # we pull the image to be sure we're scanning the latest sha available
+      - name: Pull Latest Image
+        run: docker pull ${{ matrix.image }}:${{ env.tag }}
+
+      - name: Run Trivy Vulnerability Scanner
+        uses: aquasecurity/trivy-action@8bd2f9fbda2109502356ff8a6a89da55b1ead252 # 0.9.1
+        with:
+          image-ref: ${{ matrix.image }}:${{ env.tag }}
+          format: 'sarif'
+          ignore-unfixed: true
+          severity: HIGH,CRITICAL
+          output: 'trivy-results.sarif'
+          limit-severities-for-sarif: true
```

Comment:
We might consider removing the limits, it's probably better to handle them downstream in the Security tab either discarding them or handling them even if not fixed or lower serverity:
```suggestion
          image-ref: ${{ matrix.image }}:${{ env.tag }}
          format: 'sarif'
          output: 'trivy-results.sarif'
```